### PR TITLE
add `css.VirtualZPowGate` (qudit virtual phase gate)

### PR DESCRIPTION
**Pull request by richrines1**
similar to the existing `css.QutritZ<k>PowGate` types, but redefined in a way that extends naturally to arbitrary dimensions

will formally deprecate `css.QutritZ<k>PowGate`s after server-side updates to use this one

### CODE REVIEW

File: cirq-superstaq/cirq_superstaq/ops/qudit_gates.py [hunk 1]
```diff
@@ -236,6 +236,130 @@ def __repr__(self) -> str:
...
+        self,
+        dimension: int = 2,
+        level: int = 1,
+        exponent: cirq.TParamVal = 1.0,
+        global_shift: float = 0.0,
+    ) -> None:
+        """Initializes as `VirtualZPowGate`.
+
+        Args:
+            dimension: The qudit dimension on which this gate will act.
```
Comment (paaige) [inline cirq-superstaq/cirq_superstaq/ops/qudit_gates.py:252]: Can you add something explaining how the acronym / abbrev. relates to dimension and level? E.g. VZ3_12 not really intuitive that it means VirtualZPowGate(dimension=3, level=2)

File: cirq-superstaq/cirq_superstaq/ops/qudit_gates.py [hunk 2]
```diff
@@ -236,6 +236,130 @@ def __repr__(self) -> str:
...
+            level: The lowest energy level onto which this gate applies a phase. For example,
+                passing `level=2` a phase of `(-1)**exponent` will applied to energy levels
```
Comment (paaige) [inline cirq-superstaq/cirq_superstaq/ops/qudit_gates.py:254]: `will be`

File: cirq-superstaq/cirq_superstaq/ops/qudit_gates.py [hunk 3]
```diff
@@ -236,6 +236,130 @@ def __repr__(self) -> str:
...
+        return VirtualZPowGate(
+            dimension=self._dimension,
+            level=self._level,
+            exponent=exponent,
+            global_shift=self._global_shift,
+        )
+
+    def _eigen_components(self) -> List[Tuple[float, npt.NDArray[np.float_]]]:
+        projector_phase = np.zeros(self._dimension)
+        projector_phase[self._level :] = 1
```
Comment (paaige) [inline cirq-superstaq/cirq_superstaq/ops/qudit_gates.py:303]: `[self._level:]`?

Comment (richrines1) [reply cirq-superstaq/cirq_superstaq/ops/qudit_gates.py:303]: i thought so too but the formatter wants it this way :/

File: cirq-superstaq/cirq_superstaq/ops/qudit_gates.py (hunk 1 above)
Comment (richrines1) [reply cirq-superstaq/cirq_superstaq/ops/qudit_gates.py:252]: ended up changing all the aliases to e.g. `QutritVZ2` (for dimension 3, level 2) - lmk if you think they're any more clear

**Review by paaige — APPROVED** [review]
lgtm!

**Pull request merged**